workspaceitem-data-cclicense.md

```diff
@@ -0,0 +1,84 @@
+# WorkspaceItem data of CC License sectionType
+[Back to the definition of the workspaceitems endpoint](workspaceitems.md)
+
+The section data represents the data about the CC license
+
+```json
+{
+  	"granted": true,
+  	"uri": "http://creativecommons.org/licenses/by-nc-sa/3.0/us/",
+  	"rights": "Attribution-NonCommercial-ShareAlike 3.0 United States",
+	"file":  
+  	 	{
+          "uuid": "1ce6db0e-662f-4a13-ba87-c371ad664b14",
+          "name": "license_rdf",
+          "handle": null,
+          "metadata": {
+              "dc.title": [
+                {
+                  "value": "license_rdf",
+                  "language": null,
+                  "authority": null,
+                  "confidence": -1,
+                  "place": 0
+                }
+              ]
+          },
+          "sizeBytes": 1012,
+          "checkSum": {
+            "checkSumAlgorithm": "MD5",
+            "value": "59a4cfd819ce67ba7252efc52d1eb99b"
+          },
+          "sequenceId": 1,
+          "type": "bitstream"
+        }
+}
+```
+
+The following attributes can be defined:
+* granted: true if a CC License has been specified
+* uri: the URI of the CC License, typically stored in dc.rights.uri, can be null
+* rights: the rights name of the CC License, typically stored in dc.rights, can be null
+* file: the bitstream containing the license, can be null
+
+## Patch operations
+The PATCH method expects a JSON body according to the [JSON Patch specification RFC6902](https://tools.ietf.org/html/rfc6902)
+
+Each successful Patch operation will return a HTTP 200 CODE with the new workspaceitem as body
+
+### Add
+To accept a CC license the client must send a JSON Patch ADD operation to the *granted* path
+
+```json
```

Comment:
this doesn't follow the json patch specification. The structure of the value send in the patch must respect the structure of the section data, ie you cannot use arbitrary json path here such as license, fields etc that will be not reflected in the updated json.

I suggest to just sent the value for the URI that should be retrieved interacting with the configuration endpoint and leave all the other attribute in the section data as readonly.
There is no need for a granted boolean attribute here as you can add / remove the uri attribute as said above. For the normal license was different as the other attribute is the timestamp and we don't want the client to be able to specify the timestamp for such reason the client was needed to manipulate the granted attribute

Reply:
Summary of what was discussed in the meeting:
* A search can be offered at /api/config/submissioncclicenses/search/rightsByQuestions which accepts the license name and the answers to all questions
* This search will return the dc.rights.uri, dc.rights and contents for the license_rdf
* Another search can be offered at /api/config/submissioncclicenses/search/rdfByQuestions which accepts the license name and the answers to all questions
* This search will return the license_rdf file contents
* One PATCH request will add the dc.rights.uri, dc.rights
* Another POST request will add the license_rdf (since it's an upload)
* REST will need to validate whether dc.rights.uri, dc.rights are "sensible" for a CC license

Please update this summary if you see anything the differs from what we discussed

I will verify how much work this would require in REST and Angular.

Reply:
@tdonohue when having to implement this solution, I would expect both REST and Angular's estimates to be quite tight. Perhaps the estimates should be increased with a day on both parts to be safe with these specs

Reply:
> A search can be offered at /api/config/submissioncclicenses/search/rightsByQuestions which accepts the license name and the answers to all questions

OK

> This search will return the...

you only need to get the license uri, i.e. http://creativecommons.org/licenses/by-nc-sa/3.0/us/

> Another search can be offered at /api/config/submissioncclicenses/search/rdfByQuestions which accepts the license name and the answers to all questions
> ...

Nothing else is needed. You don't need to retrieve the RDF from the configuration once that the license will be added patching the workspace with the add of the license uri the server side will import the rdf from CC and add it as a bitstream that can be exposed in readonly in the section. 
Only the uri field is writable, no need for complex validation in such way. If you add a value the specified license will be added if you remove the uri the license and the related bitstream will be dropped

Reply:
It sounds like @abollini is suggesting a simplification of what we need to do to implement this feature. If we can simply wrap the Creative Commons API, we should only need to return the License URI (e.g. http://creativecommons.org/licenses/by-nc-sa/3.0/us/) from our implementation. Everything else from CC can be obtained via that URL.  So, if we find a need for the RDF, it's available easily by appending `/rdf` on that URI, e.g. http://creativecommons.org/licenses/by-nc-sa/3.0/us/rdf.  

So, I think I agree that I don't see a need to over complicate this.  Let's just build a simple way to obtain the proper License URI from the CC API.  From that URI, we can always use the CC API again to pull down more info, or the RDF version, if necessary.

We can talk about this more in tomorrow's meeting if necessary.